flycheck.el

```diff
@@ -4095,19 +4140,29 @@ refresh the mode line."
 
 STATUS defaults to `flycheck-last-status-change' if omitted or
 nil."
-  (let ((text (pcase (or status flycheck-last-status-change)
-                (`not-checked "")
-                (`no-checker "-")
-                (`running "*")
-                (`errored "!")
-                (`finished
-                 (let-alist (flycheck-count-errors flycheck-current-errors)
-                   (if (or .error .warning)
-                       (format ":%s|%s" (or .error 0) (or .warning 0))
-                     "")))
-                (`interrupted ".")
-                (`suspicious "?"))))
-    (concat " " flycheck-mode-line-prefix text)))
+  (let (text
+        face)
+    (setq text (concat " " flycheck-mode-line-prefix
+                       (pcase (or status flycheck-last-status-change)
```

Comment:
Can you move the `pcase` out of the `concat`, as I think the way it's current written the readability is not optimal? (plus it shift the indentation too much to the right)